app/Domain/Tokens/Token.h

```diff
@@ -0,0 +1,29 @@
+//
+// Created by Gabriel Noriega on 2018-02-15.
+//
+
+#ifndef SMALLWORLD_TOKEN_H
+#define SMALLWORLD_TOKEN_H
+
+#include "TokenType.h"
+#include "TokenSubtype.h"
+
+struct Token {
+    TokenType const m_type;
+    TokenSubtype const m_subtype;
+    bool m_inDecline{false};
+    bool m_transient{false};
+
+    Token(TokenType v_type, TokenSubtype v_subtype) : m_type(v_type), m_subtype(v_subtype) {}
+
+    ~Token() = default;
```

Comment:
Can you explain this to me? :)

Reply:
Line 17 is the constructor if that's what you're asking. This is the C++11 way if I'm not mistaken.
Basically is doing this in a cleaner way.
``` C++
Token (TokenType v_type, TokenSubtype v_subtype) {
  m_type = v_type;
  m_subtype = v_subtype;
}
```

Reply:
Alright, good to know. I didn't know about it.

Reply:
The `~Token() = default;` is just the destructor, but the default since there is actually no pointers in the class.